### DISCUSSION TITLE
Get host name and DB name from log_connection_yield

### DISCUSSION THREAD

**Discussion by skrystosik** [jeremyevans/sequel#2132] [General]
Hi, is there any way to get database name and server / host name from extension that is overwriting this method? Can't find anything interesting by inspecting objects. 
Sebastian

**Comment by jeremyevans** [comment]
If you aren't using sharding, you can inspect `opts[:database]` and `opts[:host]`.  If you are using sharding, you would have to maintain a cache keyed by connection object and mapping to the server/shard, and use the server/shard info to find the information in the Database's `:servers` hash.

**Reply by skrystosik** [reply]
thx, I’ve done inspect of opts and there was no such things. For sqlite I need at least file name of db. I don’t have on test environment any normal db at the moment

**Reply by jeremyevans** [reply]
SQLite is not a client/server database, so it would not have a `:host` option.  It should have a `:database` option, though, unless you are using a memory database.

**Reply by skrystosik** [reply]
My bad, this was in memory instance. Thx :)

**Reply by skrystosik** [reply]
I have another question, this extension works fine when I create connection manually in padrino controller code. When I use padrino with ORM (it uses sequel in this scenario) log_connection_yield is not fired. I first define ::Sequel::Database.extension in my gem. This fires model with exact DB extension and ::Sequel::Database.register_extension. 

I've seen info in doc that extensions is triggered after creating DB object, I'm not sure if I can control this with padrino. It's request from customer and can't figure this out.

**Reply by jeremyevans** [reply]
`::Sequel::Database.extension` should apply the extension to `Sequel::Database` objects created after.  However, it will not affect any already created `Sequel::Database` objects.  You could look in `Sequel::DATABASES` for existing `Sequel::Database` objects, and apply the extension to them.

Note that it is considered bad form to use `::Sequel::Database.extension` in a gem.  You should not be forcing users who load the gem to apply the extension to all databases.  You should be letting the users who load the gem choose which Database objects they want to load the extension into.  If you must use  `::Sequel::Database.extension` in your gem, do not make it the default behavior, either require a separate file or call a method for it.

**Reply by skrystosik** [reply]
It’s apm agent so rather there is no option :) It’s possible to disable instrumentation. Thx for hint
